src/components/core/ContextSelector/index.tsx

```diff
@@ -1,177 +1,139 @@
     };
 
-    const onSearchChange = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setQueryText(e.target.value);
-    }, []);
-
-    const onOpen = React.useCallback(() => setShouldShowResultsDropdown(true), []);
-    const onClose = React.useCallback(() => {
-        setShouldShowResultsDropdown(false);
-        setQueryText('');
-    }, []);
-
-    const onKeyUp = React.useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
-        if (e.keyCode === 27) {
-            // ESC
-            onClose();
-        }
-    }, []);
-
-    React.useEffect(() => {
-        if (shouldShowResultsDropdown && inputRef) {
-            inputRef.focus();
-        }
-    }, [shouldShowResultsDropdown, inputRef]);
-
-    const containerRef = React.useRef<HTMLDivElement>(null);
-
-    const isQuerying = useContextSearchResults(
-        inputRef,
-        containerRef,
-        types,
-        relatedContexts,
-        setCurrentContextAsync,
-        queryText,
-        shouldShowResultsDropdown,
-        onClose
-    );
-
-    const inputClassNames = classNames(styles.searchInput, {
-        [styles.focus]: shouldShowResultsDropdown,
+    const dropdownController = useDropdownController((ref, isOpen, setIsOpen) => {
+        const selectedItem = React.useMemo(() => {
+            const mergedItems = mergeDropdownSectionItems(dropdownSections);
+            const selectedItem = mergedItems.find(option => option.isSelected === true);
+            return selectedItem;
+        }, [dropdownSections]);
+
+        const selectedValue = React.useMemo(() => {
+            if (isOpen) {
+                return queryText;
+            } else if (selectedItem) {
+                return selectedItem.title;
+            } else if (currentContext) {
+                return currentContext.title;
+            }
+            return '';
+        }, [isOpen, queryText, selectedItem, currentContext]);
+
+        return (
+            <>
+                <SearchIcon color="#DADADA" />
+                <input
+                    type="text"
+                    value={selectedValue}
+                    onChange={e => setQueryText(e.target.value)}
+                    onClick={() => !isOpen && setIsOpen(true)}
+                    placeholder={selectedValue !== '' ? selectedValue : 'Search contexts'}
+                    className={styles.searchInput}
+                    ref={inputRef}
+                />
+                {isQuerying && <Spinner inline />}
+            </>
+        );
     });
 
-    const getButtonContent = () => {
-        if (currentContextForType) {
-            return currentContextForType.title;
-        }
-
-        if (isFetchingRelatedContexts) {
-            return (
-                <span>
-                    Resolving {currentContext ? currentContext.title : null} <Spinner inline />
-                </span>
-            );
-        }
+    const updateOptions = (item: SearchableDropdownSection) =>

```

Comment:
This does nothing? It maps the dropdown sections and returns it, but no one ever uses the return value

Reply:
I am unsure what this is.
Might be able to just remove it. It is a part of the ContextSelector in Pro-Org. 
Its just copy paste from there. 
Looks like it just does something in place and nothing else. 
Maybe there should be a call to setDropDownSections here.

You wrote this a couple months ago, maybe you remeber the intent here ?

Reply:
Don't think i wrote this. I created the PR so the merge commit is by my user :P It's safe to remove.